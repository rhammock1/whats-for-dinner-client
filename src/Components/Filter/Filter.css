.filters {
  width: 75%;
  height: auto;
  margin: auto;
  padding: 5px 10px;
  background-color: #313753;
  border-radius: 5px;
  box-shadow: 2px 3px 6px 3px #121212; 
}
select {
  background-color: #1A1A2E;
  color: #efefef;
  border: none;
  padding: 2px
}
.filter-button {
  background-color: #16213E;
  color: #efefef;
  border: none;
  border-radius: 10%;
  padding: 5px 7px;
  box-shadow: 1px 2px 1px #121212; 
}
.form-group {
  display: flex;
  flex-direction: column;
  padding: 10px 0px;
}
fieldset {
  border: 0px;
}