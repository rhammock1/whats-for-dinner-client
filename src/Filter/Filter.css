.filters {
  width: 75%;
  height: auto;
  margin: auto;
}
.form-group {
  display: flex;
  flex-direction: column;
  padding: 10px 0px;
}
fieldset {
  border: 5px solid;
}