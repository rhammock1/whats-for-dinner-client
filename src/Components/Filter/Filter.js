/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import Context from '../../Context';
import './Filter.css';

class Filter extends React.Component {
  static contextType = Context;

  render() {
    const { handleWheelOptions, inOrOut } = this.context;
    const { handleChange } = this.props;
    return (
      <form onSubmit={(e) => handleWheelOptions(e)} onChange={handleChange} className="filters">
        <fieldset>
          <legend><h2>What are you looking for?</h2></legend>
          <div className="form-group">
            <label htmlFor="inOrOut">Restaurants or Recipes?</label>
            <select id="inOrOut" name="inOrOut" required>
              <option value="">Choose One:</option>
              <option value="restaurants">Restaurants</option>
              <option value="recipes">Recipes</option>
              <option value="both">Either</option>
            </select>
          </div>
          {(inOrOut === 'restaurants' || inOrOut === 'both')
            ? (
              <div className="form-group">
                <label htmlFor="style">Style: </label>
                {' '}
                <select id="style" name="style" required>
                  <option value="" />
                  <option value="chain">Chain</option>
                  <option value="local">Local</option>
                </select>
                {' '}

              </div>
            )
            : ''}
          <button className="filter-button" type="submit">Submit</button>

        </fieldset>
      </form>
    );
  }
}

export default Filter;
