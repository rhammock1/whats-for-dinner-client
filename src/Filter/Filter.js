import React from 'react';
import Context from '../Context';
import './Filter.css';

class Filter extends React.Component {
 
  static contextType = Context;

  render() {
    return (
      <form onSubmit={e => this.context.handleWheelOptions(e)}onChange={this.props.handleChange} className='filters'>
      <fieldset>
        <legend>What are you looking for?</legend>
        <div className='form-group'>
        <label htmlFor='inOrOut'>Restaurants or Recipes?</label>
        <select id='inOrOut' name='inOrOut'>
          <option value='null'>Choose One:</option>
          <option value='restaurants'>Restaurants</option>
          <option value='recipes'>Recipes</option>
          <option value='both'>Either</option>
        </select>
        </div>
        {(this.context.inOrOut === 'restaurants' || this.context.inOrOut === 'both')
          ? <div className='form-group'><label htmlFor='style'>Style: </label> <select id='style' name='style' required>
            <option value='null'></option>
            <option value='chain'>Chain</option>
            <option value='local'>Local</option>
          </select> </div>
          : ''
        }
        <button type='submit'>Submit</button>
        
      </fieldset>
    </form>
      )
    
  }
}

export default Filter;