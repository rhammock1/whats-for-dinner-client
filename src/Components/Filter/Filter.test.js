import React from 'react';
import ReactDOM from 'react-dom';
import Filter from './Filter';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Filter component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter><Filter />
    </BrowserRouter>, div)
  })
  it('renders the complete form', () => {
    const wrapper = shallow(<Filter />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})