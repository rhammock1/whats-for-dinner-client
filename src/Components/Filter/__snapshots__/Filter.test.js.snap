// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders the complete form 1`] = `
<form
  className="filters"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      <h2>
        What are you looking for?
      </h2>
    </legend>
    <div
      className="form-group"
    >
      <label
        htmlFor="inOrOut"
      >
        Restaurants or Recipes?
      </label>
      <select
        id="inOrOut"
        name="inOrOut"
        required={true}
      >
        <option
          value=""
        >
          Choose One:
        </option>
        <option
          value="restaurants"
        >
          Restaurants
        </option>
        <option
          value="recipes"
        >
          Recipes
        </option>
        <option
          value="both"
        >
          Either
        </option>
      </select>
    </div>
    <button
      className="filter-button"
      type="submit"
    >
      Submit
    </button>
  </fieldset>
</form>
`;
